core/src/main/java/org/jdbi/v3/ExtensionRegistry.java

```diff
@@ -31,34 +31,38 @@ static ExtensionRegistry copyOf(ExtensionRegistry registry) {
         return new ExtensionRegistry(registry.factories);
     }
 
-    private static class Entry {
-        static Entry copyOf(Entry entry) {
-            return new Entry(entry.factory, entry.config.createCopy());
+    private static class Entry<C extends ExtensionConfig<C>> {
+        static <C extends ExtensionConfig<C>> Entry<C> copyOf(Entry<C> entry) {
+            return new Entry<>(entry.factory, entry.config.createCopy());
         }
 
-        final ExtensionFactory factory;
-        final ExtensionConfig config;
+        final ExtensionFactory<C> factory;
+        final C config;
 
-        Entry(ExtensionFactory factory, ExtensionConfig config) {
+        Entry(ExtensionFactory<C> factory, C config) {
             this.factory = factory;
             this.config = config;
         }
+
+        <E> E attach(Class<E> extensionType, Supplier<Handle> handle) {
+            return factory.attach(extensionType, config, handle);
+        }
     }
 
-    private final List<Entry> factories;
+    private final List<Entry<? extends ExtensionConfig<?>>> factories;
```

Comment:
Thankfully this is all private so we can change it later if necessary.  I hate warnings more than I hate extra types just to satisfy the checker :)